### PR TITLE
Complete tests for 100% coverage and bump release for 1.0.0

### PR DESCRIPTION
**Pull request by altusgerona**

### CODE REVIEW

File: pyrmq/consumer.py [hunk 1]
```diff
@@ -131,7 +154,7 @@ def close(self) -> None:
         """
         Manually closes a connection to RabbitMQ. Useful for debugging and tests.
         """
-        self.connection.close()
+        self.thread.join(0.1)
```
Comment (sibayanjasper) [inline pyrmq/consumer.py:157]: ?

File: pyrmq/publisher.py [hunk 2]
```diff
@@ -22,11 +22,13 @@
 from pika.adapters.blocking_connection import BlockingChannel
 from pika.exceptions import AMQPConnectionError, AMQPChannelError, StreamLostError
 from pika.spec import PERSISTENT_DELIVERY_MODE
-
+import logging
```
Comment (sibayanjasper) [inline pyrmq/publisher.py:25]: Organize imports

File: pyrmq/publisher.py [hunk 3]
```diff
@@ -104,6 +114,7 @@ def declare_queue(self, channel) -> None:
             exchange=self.exchange_name,
             routing_key=self.routing_key,
         )
+        channel.confirm_delivery()
```
Comment (sibayanjasper) [inline pyrmq/publisher.py:117]: ?

File: pyrmq/consumer.py (hunk 1 above)
Comment (altusgerona) [reply pyrmq/consumer.py:157]: So there's no way to manually close a thread in Python, right? But rerunning the tests results in flaky behavior which I think stems from opened connections by pika. Will have to investigate further. In any case, one way to "stop" it is by joining it back to the main thread. This essentially ends the consumption and allows better cleanup per test. I'm still thinking whether this should be part of the global teardown but I figure it's better to leave here so devs can see that multiple consumers should be closed to maintain consistency in dealing with a single queue or exchange.

File: pyrmq/publisher.py (hunk 3 above)
Comment (altusgerona) [reply pyrmq/publisher.py:117]: According to [this](https://pika.readthedocs.io/en/stable/examples/blocking_publish_mandatory.html) it's a way to assure that messages are delivered. Personally I didn't see any difference in my tests but YMMV. We can remove this if you want.

File: pyrmq/consumer.py (hunk 1 above)
Comment (sibayanjasper) [reply pyrmq/consumer.py:157]: Why 0.1?

File: pyrmq/publisher.py (hunk 3 above)
Comment (sibayanjasper) [reply pyrmq/publisher.py:117]: Okay, we can leave this.

File: pyrmq/consumer.py (hunk 1 above)
Comment (altusgerona) [reply pyrmq/consumer.py:157]: It's the timeout value. Could be anything really. Just tried 0.1 seconds to make it resolve as fast as possible. There might be a better way.

Comment (sibayanjasper) [reply pyrmq/consumer.py:157]: Okay, we can leave it for now. Besides, we're only using this for tests.

**Pull request merged**